# Nanopore sequencing of internal 2′-PO4 modifications installed by RNA repair

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.7604091.svg)](https://doi.org/10.5281/zenodo.7604091)

Ligation by plant and fungal RNA ligases yields an internal 2′-phosphate group on each RNA ligation product. Here, we define several unique signals produced by 2′-phosphorylated RNAs during nanopore sequencing:

* A 2′-phosphate at the splice junction of HAC1 mRNA inhibits 5′→3′ degradation, enabling detection of decay intermediates by 5′ end mapping in yeast RNA repair mutants.
* During direct RNA sequencing, intact 2′-phosphorylated RNAs produce diagnostic changes in nanopore current properties and base calling features.

This repository describes our general bioinformatic strategy for detecting 2′-phosphate signals, as well as R markdown files used to generate the figures.

Alignment references used in this work were:

* An [S. cerevisiae transcriptome reference generated by the Jacobson Lab](https://github.com/Jacobson-Lab/yeast_transcriptome_v5), which contains both pre- and post-spliced mRNA references, with 5´- and 3´-UTRs.

* A custom _S. cerevisiae_ tRNA reference containing a consensus sequence for each cytoplasmic tRNA species in budding yeast, built from tRNAscan-SE gene predictions available at [gtRNAdb](http://gtrnadb.ucsc.edu). This reference is specific to the sequencing library preparation described in in [PMID: 34618430](https://pubmed.ncbi.nlm.nih.gov/34618430/), with tRNA splint adapter sequences appended and prepended to enable alignment to both mature tRNA sequence and ligated adapters. All tRNAs in this reference have had CCA added to their 3′ ends, and (where applicable) introns removed in silico.

* 18S and 25S rRNA fastas for reanalysis of dwell time at select sites in [PMID: 35252946](https://pubmed.ncbi.nlm.nih.gov/35252946/) were obtained from the authors' [RNA_nanoSHAPE](https://github.com/physnano/rRNA_nanoSHAPE) repository

* The RNA oligonucleotide sequences in [PMID: 34893601](https://pubmed.ncbi.nlm.nih.gov/34893601/) share a common sequence, but are decorated with different collections of RNA modifications. Their sequence was extracted from the manuscript and can be found here as **oligos.fa**.
* A reference for the synthetic RNA oligonucleotide generated by the splint ligation strategy in Figure S4 is contained in **splint.fa**

Data required to replicate the figures:

* Raw data (fast5s and fastqs) from this work are [hosted on SRA](https://www.ncbi.nlm.nih.gov/sra?linkname=bioproject_sra_all&from_uid=910992) and are associate with BioProject accession number [PRJNA910992](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA910992).

* Smaller intermediate files used to replicate the figures (e.g., bedgraphs) can be found in `/data`

* Summary files produced during MinKNOW runs are quite large; we have trimmed these file to the first 10000 reads but are happy to provide complete files upon request. The end status field we interrogate in these in these summary files is also duplicated within the raw fast5 files.

* To analyze current and dwell time on a per read level, raw nanopore sequencing data was analyzed with [Nanopolish](https://github.com/jts/nanopolish) eventalign (see src/nanopolish.sh). This per-read, per-nucleotide information generates large tabular data, which we have trimmed and hosted on Zenodo for file size constraint reasons under DOI [10.5281/zenodo.7604091](https://doi.org/10.5281/zenodo.7604091).
